src/main/java/Response/Response.java

```diff
@@ -38,6 +45,18 @@ public Builder messageBody(String messageBody) {
     public Response build() {
       return new Response(this);
     }
+
+    private int parseStatusCode(String statusCodeAndReason) {
+      int startOfStatusCode = 0;
```

Comment:
It's a bit strange to me that you are parsing the status code and reason phrase like this. Why not have a `HttpStatusCode` class that provides this for you? Something like:

```java
public final class HttpStatusCode {
  public static final int OK = 200;
  public static final int NOT_FOUND = 404;
  public static final int METHOD_NOT_ALLOWED = 405;

  private static final Map<int, String> messages = Map.ofEntries(
      Map.entry(OK, "OK"),
      Map.entry(NOT_FOUND, "OK"),
      Map.entry(METHOD_NOT_ALLOWED, "OK")
  );

  public static final String getStatusMessage(int statusCode) {
    return messages.get(statusCode);
  }
}
```

Reply:
I like this idea a lot. I was concerned about this approach before I realized that the static variables were public. Otherwise, I felt like it was going to lead me to build Response objects using magic numbers (e.g. `new Response.Builder(200).build()`). I really like that the variables allow me to never have to build a Response using an integer directly, which makes the building so much easier to read! Thank you.